crates/bevy_ecs/src/reflect.rs

```diff
@@ -27,7 +27,36 @@ pub struct ReflectComponent {
     copy: fn(&World, &mut World, Entity, Entity),
 }
 
+/// The raw functions needed to make up a [`ReflectComponent`].
+/// 
+/// This is used when creating custom implementations of [`ReflectComponent`] with
+/// [`ReflectComponent::new()`].
+pub struct ReflectComponentFns {
```

Comment:
Should these structs have their own `ReflectComponentFns::new<T: Component>() -> Self` functions to fill in the entire struct, allowing individual modification if needed? Or does it not make sense to only override some but not all function pointers?

Reply:
That's actually a good idea. I'm not 100% sure that there will be a use for overriding only some options, but it seems like it could happen.